File: frontend/ilhan-store-client/src/types/Category.ts
export interface Category {
  id: number
  name: string
  description?: string | null
  imageUrl?: string | null
  isMainCategory: boolean
  parentCategoryIds: number[]
  productCount: number
}

export interface CreateCategoryRequest {
  name: string
  description?: string | null
  imageUrl?: string | null
  isMainCategory: boolean
  parentCategoryIds: number[]
}

export type UpdateCategoryRequest = CreateCategoryRequest

File: backend/IlhanStore.Entity/Entities/Category.cs
namespace IlhanStore.Entity.Entities;

public class Category : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsMainCategory { get; set; } = true;

    // Relationships where this category is the PARENT (its child subcategories)
    public ICollection<CategoryRelationship> ChildRelationships { get; set; } = [];
    // Relationships where this category is the CHILD (its parent categories)
    public ICollection<CategoryRelationship> ParentRelationships { get; set; } = [];
    public ICollection<Product> Products { get; set; } = [];
}
